### PR TITLE
ci: Switch main CI job to Ubuntu 22.04 

### PR DESCRIPTION
**Pull request by paulgessinger**
I'm adding the old Ubuntu 20.04 job as an "extra" job, basically switching it out for the prior 22.04 one.

### CODE REVIEW

**Comment by paulgessinger** [comment]
Physmon and python level tests still fail, likely due to various version changes.

**Comment by paulgessinger** [comment]
IVF seeded:
diffz
image

IVF truth estimated:
diffz
image
recoz
image

IVF orthogonal:
recoz
image

GSF truth tracking:
resmean t vs eta
image


---

I think all of these changes are tolerable.

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -279,7 +279,7 @@ jobs:
           && git config --global safe.directory "$GITHUB_WORKSPACE"
           && pip3 install histcmp==0.5.2
           && pip3 install -r Examples/Scripts/requirements.txt
-          && /usr/local/bin/download_geant4_data.sh
+          && /usr/local/bin/geant4-config --install-datasets
```
Comment (paulgessinger) [inline .github/workflows/builds.yml:282]: Turns out G4 actually ships with a script that does exactly what we need here. 😅

**Comment by benjaminhuth** [comment]
I don't understand that GSF change. Seems also to happen in #1951

**Comment by benjaminhuth** [comment]
> I don't understand that GSF change. Seems also to happen in #1951

Okay, I see the same locally on `main`... not sure what could be the reason, I'll have a look at it.

**Review by benjaminhuth — APPROVED** [review]
looks good to me!

**Pull request merged**